mkosi/distributions/postmarketos.py

```diff
@@ -0,0 +1,105 @@
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "postmarketOS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "ext4"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.apk
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "edge"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.postmarketos
+
+    @classmethod
+    def package_manager(cls, config: Config) -> type[PackageManager]:
+        return Apk
+
+    @classmethod
+    def setup(cls, context: Context) -> None:
+        Apk.setup(context, list(cls.repositories(context)))
+
+        # TODO: Create merged /usr manually for now until our upstream (Alpine Linux) supports it:
+        # https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/85504
+        for dir in ["lib", "bin", "sbin"]:
+            (context.root / "usr" / dir).mkdir(parents=True, exist_ok=True)
+            (context.root / dir).symlink_to(f"usr/{dir}")
+
+        with complete_step("Setting up postmarketOS keyring"):
+            # Create keys directory in sandbox
+            keys_dir = context.sandbox_tree / "etc/apk/keys"
+            keys_dir.mkdir(parents=True, exist_ok=True)
+
+            # Copy keys from various sources (if they exist)
+            for d in [
+                context.config.tools() / "usr/lib/apk/keys",
+                context.config.tools() / "usr/share/distribution-gpg-keys/alpine-linux",
+                context.config.tools() / "usr/share/distribution-gpg-keys/postmarketos",
+            ]:
+                if d.exists():
+                    copy_tree(d, keys_dir, sandbox=context.sandbox)
+
+    @classmethod
+    def install(cls, context: Context) -> None:
+        Apk.install(context, ["postmarketos-baselayout", "postmarketos-release"], apivfs=False)
+
+    @classmethod
+    def repositories(cls, context: Context) -> Iterable[ApkRepository]:
+        if context.config.release != "edge":
+            die(f"Only 'edge' release is currently supported, got '{context.config.release}'")
+
+        if context.config.local_mirror:
+            yield ApkRepository(url=context.config.local_mirror)
+            return
+
+        # Alpine repos
+        for repo_name in ["main", "community", "testing"]:
+            yield ApkRepository(
+                url=f"https://dl-cdn.alpinelinux.org/alpine/{context.config.release}/{repo_name}"
+            )
+
+        # postmarketOS repos
+        mirror = context.config.mirror or "https://mirror.postmarketos.org/postmarketos"
+        subdir = "master" if context.config.release == "edge" else f"v{context.config.release}"
+
+        # systemd repo
+        url = f"{mirror}/extra-repos/systemd/{subdir}"
+        yield ApkRepository(url=url)
+
+        # main repo
+        url = f"{mirror}/{subdir}"
+        yield ApkRepository(url=url)
```

Comment:
Hmm, if from the view of mkosi postmarketos is nothing more than a repository of extra packages to use with alpine, I wonder if we should call the distribution in mkosi `alpine` and just enable the postmarketos repository with `Repositories=postmarketos` instead.

Reply:
Given all the work @yuwata is doing on musl support, it does't seem too far fetched to consider that alpine might package systemd in the future.

Though actually, we currently install postmarketos-baselayout and postmarketos-release as the base packages which also wouldn't work anymore if we made this change. So if those packages are expected to stay indefinitely, maybe having `postmarketos` as a separate distribution is the way to go after all.

Reply:
Sure let's keep it the way it is, we can support alpine later as a separate distribution and have postmarketos inherit from the alpine class when that happens.